tests/testthat/test-oauth.R

```diff
@@ -148,4 +148,222 @@ with_mock_api({
+    )
+  })
+
+  test_that("we can retrieve the oauth user credentials with audience and req token type", {
+    client <- MockConnect$new("2025.07.0")
+    client$mock_response(
+      "POST",
+      v1_url("oauth", "integrations", "credentials"),
+      content = list(
+        access_token = "user-access-token",
+        issued_token_type = "urn:ietf:params:oauth:token-type:access_token",
+        token_type = "Bearer"
+      )
+    )
+
+    credentials <- get_oauth_credentials(
+      client,
+      user_session_token = "user-session-token",
+      requested_token_type = "urn:ietf:params:oauth:token-type:access_token",
+      audience = "audience"
+    )
+    expect_equal(
+      credentials,
+      list(
+        access_token = "user-access-token",
+        issued_token_type = "urn:ietf:params:oauth:token-type:access_token",
+        token_type = "Bearer"
+      )
+    )
+  })
+
+  test_that("we can retrieve the oauth content credentials with audience", {
+    client <- MockConnect$new("2025.07.0")
+    client$mock_response(
+      "POST",
+      v1_url("oauth", "integrations", "credentials"),
+      content = list(
+        access_token = "content-access-token",
+        issued_token_type = "urn:ietf:params:oauth:token-type:access_token",
+        token_type = "Bearer"
+      )
+    )
+    credentials <- get_oauth_content_credentials(
+      client,
+      content_session_token = "content-session-token",
+      audience = "audience",
+    )
+    expect_equal(
+      credentials,
+      list(
+        access_token = "content-access-token",
+        issued_token_type = "urn:ietf:params:oauth:token-type:access_token",
+        token_type = "Bearer"
+      )
+    )
+  })
+
+  test_that("oauth viewer credentials retrieval with audience on unsupported version fails", {
+    client <- MockConnect$new("2025.06.0")
+
+    expect_error(
+      get_oauth_credentials(
+        client,
+        user_session_token = "user-session-token",
+        audience = "audience"
+      ),
+      "ERROR: This feature requires Posit Connect version 2025.07"
+    )
+  })
+
+  test_that("we can retrieve the oauth content credentials with audience and req token type", {
+    client <- MockConnect$new("2025.07.0")
+    client$mock_response(
+      "POST",
+      v1_url("oauth", "integrations", "credentials"),
+      content = list(
+        access_token = "content-access-token",
+        issued_token_type = "urn:ietf:params:oauth:token-type:access_token",
+        token_type = "Bearer"
+      )
+    )
+
+    credentials <- get_oauth_content_credentials(
+      client,
+      content_session_token = "content-session-token",
+      requested_token_type = "urn:ietf:params:oauth:token-type:access_token",
+      audience = "audience"
+    )
+    expect_equal(
+      credentials,
+      list(
+        access_token = "content-access-token",
+        issued_token_type = "urn:ietf:params:oauth:token-type:access_token",
+        token_type = "Bearer"
+      )
+    )
+  })
```

Comment:
I don't know that this test and e.g. the test on line 207 really exercise different parts of the code (same for the other tests that follow this format). Because `MockConnect` doesn't enforce expectations about the contents of the request body, we're basically confirming that the correct route is called and the expected data is returned.

I personally think it'd be worth adding that capability in the future, as for tests like this, I find it easier to have the expectations defined alongside the tests than in JSON files that are loaded implicitly from the requests.

Perhaps it makes sense to simplify these tests for now, removing a few of them.

It might also make sense to create an issue to track adding the ability to provide an expected request body in `MockConnect$mock_response`.